inst/rmd/h/default.html

```diff
@@ -52,6 +52,13 @@
 <style type="text/css">
 $highlighting-css$
 
+$if(ispandoc2)$
+pre.sourceCode  {
+  color: inherit;
+  background-color: transparent;
```

Comment:
Actually this will break themes that don't have a background color, such as `pygments` https://github.com/rstudio/rmarkdown/issues/654#issuecomment-380234085 Such themes should inherit the background color from Bootstrap's `pre`.

My head is starting to hurt...

Reply:
Yes it was not clear to me and I wanted to test it further. Thanks for the confirmation. This [blog post](https://www.garrickadenbuie.com/blog/2018/07/06/pandoc-syntax-highlighting-examples/) shows the result of pandoc styling, and indeed pygment as a white background, but with bootstrap it is not the case. As we need to 
* override the `pre` style of bootstrap because pandoc apply the background color and color to the parent
* inherits for the `pre` style of bootstrap to keep the `pre` css rule from bootstrap for theme that don't have background and color
I think we should also do something conditional. 

Of pandoc highlighting style, we have frome the [src code](https://github.com/jgm/skylighting/blob/master/skylighting-core/src/Skylighting/Styles.hs)
*  `pygments` with nothing ➡️ should inherit boostrap _color_ and _background-color_
* `tango` with no default color ➡️  should inherit boostrap color only
*  `haddock` is like `pygments`
*  `monochrome` is like `pygments`

I'll come up with something... 🤔 and yes 🤯

Reply:
I made a proposition by manually setting which style has background and color or not. Then with two more template variable, the correct rule is activated. 

In the same idea, with pandoc2 we can do something more dynamic. I hesitate to introduce this in rmarkdown but it could be interesting. We can get highlight information with 
```r
  if (pandoc2.0()) {
    arg <- c("--print-highlight-style", pandoc_highlight)
    json_res <- system2(pandoc(), arg, stdout = TRUE)
    properties <- jsonlite::fromJSON(paste0(json_res, collapse = "\n"))
  }
```

we then get for _pygments_
```r
> properties$`background-color`
NULL
> properties$`text-color`
NULL
```

and for _tango_
```r
> properties$`text-color`
NULL
> properties$`background-color`
[1] "#f8f8f8"
```

We could introduce a function `get_higligthing_styles` in rmarkdown (that we could export) and use this here to test if `is.null` or not. This could even mean that we could fill in the value in
`pre.sourceCode` css rule, instead of _inherits_ or _transparent_

I run this by you before pushing anything. I have a working code locally though - I can share then revert.